### PR TITLE
Improve startup logging

### PR DESCRIPTION
**Pull request by 2opremio**
### What

Log version and commit right after starting and add more detailed logging around in-memory storage initialization.

### Why

We found situations in which RPC took 30 mins to start

### Known limitations

N/A

### CODE REVIEW

File: cmd/soroban-rpc/internal/daemon/daemon.go [hunk 1]
```diff
@@ -196,7 +202,20 @@ func MustNew(cfg *config.Config) *Daemon {
...
+	var initialSeq uint32
+	var currentSeq uint32
 	err = db.NewLedgerReader(dbConn).StreamAllLedgers(readTxMetaCtx, func(txmeta xdr.LedgerCloseMeta) error {
+		currentSeq = txmeta.LedgerSequence()
+		if initialSeq == 0 {
+			initialSeq = currentSeq
+			logger.WithFields(supportlog.F{
+				"seq": currentSeq,
+			}).Info("initializing in-memory store")
+		} else if (currentSeq-initialSeq)%inMemoryInitializationLedgerLogPeriod == 0 {
```
Comment (sreuland) [inline cmd/soroban-rpc/internal/daemon/daemon.go:214]: this is great to print ongoing status, maybe the period of 1M ledgers does not get reached, is it worthwhile to use a counter that the function closure increments and uses as trigger instead?

**Review by sreuland — APPROVED** [review]
lgtm, thanks for adding this!

**Comment by sreuland** [comment]
this was triggered during investigation of https://github.com/stellar/soroban-rpc/issues/128

File: cmd/soroban-rpc/internal/daemon/daemon.go (hunk 1 above)
Comment (2opremio) [reply cmd/soroban-rpc/internal/daemon/daemon.go:214]: There are around 40M ledgers in pubnet. I don’t want to flood the logs

**Pull request merged**